eclair-core/src/main/scala/fr/acinq/eclair/io/Peer.scala

```diff
@@ -59,26 +58,39 @@ class Peer(nodeParams: NodeParams, remoteNodeId: PublicKey, authenticator: Actor
   }
 
   when(DISCONNECTED) {
-    case Event(Peer.Connect(NodeURI(_, hostAndPort)), d: DisconnectedData) =>
-      val address = new InetSocketAddress(hostAndPort.getHost, hostAndPort.getPort)
-      if (d.address_opt.contains(address)) {
-        // we already know this address, we'll reconnect automatically
-        sender ! "reconnection in progress"
-        stay
-      } else {
-        // we immediately process explicit connection requests to new addresses
-        context.actorOf(Client.props(nodeParams, authenticator, address, remoteNodeId, origin_opt = Some(sender())))
-        stay
+    case Event(Peer.Connect(_, address_opt), d: DisconnectedData) =>
+      address_opt.map(hAndp => new InetSocketAddress(hAndp.getHost, hAndp.getPort)).orElse {
+        getPeerAddressFromNodeAnnouncement()
+      } match {
+        case None =>
+          log.warning(s"Unable to connect to $remoteNodeId no address found")
+          sender() ! s"Unable to connect to $remoteNodeId no address found"
```

Comment:
```suggestion
          sender() ! "no address found"
```

Consistency with existing messages